integration/walletextension/wallet_extension_test.go

```diff
@@ -427,6 +429,25 @@ func TestCannotSubscribeOverHTTP(t *testing.T) {
 	}
 }
 
+func TestSuccessGasEstimate(t *testing.T) {
```

Comment:
I feel like we need a place for integration tests of specific methods/scenarios but maybe not as part of this PR, we should think about it. I can see why it ended up in here even though it's not really testing wallet extension, because it feels cleaner to put it here than shoehorn it into the simulation tests.

Reply:
Yeah that's a really good point.
I think as we expand on the WE and test coverage we'll be able to decouple testing WE wrapping and enclave logic ?
That's kinda where I was going with the `enclave_test.go` but as you refered we should brainstorm a bit more about it. This was just a first go at it I guess.

Reply:
Yeah - these have stopped being wallet extension tests, and more tests of single RPC calls. Maybe it would clear it up to just rename this to `rpc_test.go` or whatever, and add a readme (in another PR)?